RecoBTag/Combined/interface/hiRun2LegacyCSVv2Tagger.h

```diff
@@ -0,0 +1,56 @@
+#ifndef RecoBTag_Combined_hiRun2LegacyCSVv2Tagger_h
+#define RecoBTag_Combined_hiRun2LegacyCSVv2Tagger_h
+
+#include "FWCore/Framework/interface/ESConsumesCollector.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "CommonTools/MVAUtils/interface/TMVAEvaluator.h"
+#include "RecoBTau/JetTagComputer/interface/JetTagComputer.h"
+#include "RecoBTag/SecondaryVertex/interface/CombinedSVSoftLeptonComputer.h"
+#include "DataFormats/BTauReco/interface/TaggingVariable.h"
+#include "RecoBTau/JetTagComputer/interface/JetTagComputerRecord.h"
+
+#include <memory>
+#include <type_traits>
+
+/** \class hiRun2LegacyCSVv2Tagger
+ *  \author M. Nguyen
+ *  copied from CharmTagger.h (by M. Verzetti)
+ */
+
+class hiRun2LegacyCSVv2Tagger : public JetTagComputer {
```

Comment:
- class name should be capitalized
- what makes it Run2?
- what makes it Legacy?
- what makes it v2? 

I'd call it HeavyIonsCSVTagger or HeavyIonsCSVv1Tagger if we want to commit to versioning for simultaneous support of multiple versions.

Reply:
This is the csvV2 tagger, as defined by CMS for the outside world.   The version signifies both the training variables and the training method.  
This version of the tagger was used for the 2015 and 2018 PbPb data. 
It's now 'legacy' b/c it's based on quite outdated methods, which are not really supported in pp anymore.  It's legacy in the sense that after this reMiniAOD of the 2018 PbPb data, we will not support this for Run 3, or for any further processing of old data.   From everything we've seen so far, there should not be any need for dedicated heavy-ion b-tagging code, so I don't foresee future versions of this.

Reply:
I suppose we'll need to iterate. I'd rather have similar naming pattern to the existing one.
Also, the pp CSV is apparently in RecoBTag/SecondaryVertex.

If this tagger was used on 2015 data, is hiRun2LegacyCSVv2Tagger something that was named this way in analysis recipes?

For the old pp track-based CSV tagger
- apparently there is no V2 C++ class, just the configuration (assuming this refers to CombinedSVComputer)
- is "Legacy" really meant to reflect that this is using tracks, not candidates? Then the non-HI version comes without the extra modifiers and we better stay closer to what's available.

Reply:
Yeah, I realized after my first attempt that the pp one is under SecondaryVertex.  
But when I moved the code there I ran into an issue related to the declaration of CombinedSVSoftLeptonComputer sl_computer_;
The variable SoftLeptonFlip gets defined twice or something. 
I wasn't able to easily figure it out. 

CSVv2 is the name of the tagger, even for the world outside CMS. 
The Run2Legacy bit provides extra info, but if you don't like, how about we go for   "HeavyIonCSVv2" ?